drivers/interrupt_controller/Kconfig.plic

```
@@ -13,6 +13,44 @@ config PLIC
 
 if PLIC
 
+config PLIC_SUPPORTS_SOFT_INTERRUPT
+	bool
+	default y
+	depends on DT_HAS_ANDESTECH_NCEPLIC100_ENABLED
+	help
+	  Enabled when the PLIC supports software-triggered interrupts.
+
+config PLIC_SUPPORTS_TRIG_TYPE
+	bool
+	default y
+	depends on DT_HAS_ANDESTECH_NCEPLIC100_ENABLED
+	help
+	  Enabled when the PLIC supports multiple trigger types,
+	  such as level, edge, etc.
+
+if PLIC_SUPPORTS_TRIG_TYPE
+
+config PLIC_TRIG_TYPE_REG_OFFSET
+	hex "Trigger type register offset"
+	default 0x1080 if DT_HAS_ANDESTECH_NCEPLIC100_ENABLED
+	help
+	  Offset to the 'trigger type' register.
+
+config PLIC_TRIG_TYPE_WIDTH
+	int "Trigger type width"
```

"bitwidth" is probably a better term

```suggestion
config PLIC_TRIG_TYPE_BITWIDTH
	int "Trigger type bitwidth"
```